Marketplace: Do not show eligibility modal when not needed

epeicher opened the pull request:
The eligibility modal should not be displayed if there are no warnings
or additional messages to show

#### Proposed Changes

* Only show eligibility modal if there are messages to display

#### Testing Instructions

* Go to a **Simple** site that is eligible to install plugins. For example, a Business or Pro site that has not yet been transferred to Atomic
* Make sure the site has a domain, you can use a `.blog` domain for testing purposes
* Go to plugins and Install a Plugin, it can be either a Paid or Free plugin
* Make sure there is no additional modal displayed and the installation continues successfully or the checkout page is displayed. 

Fixes #62342

epeicher committed:
Marketplace: Do not show eligibility modal when not needed

The eligibility modal should not be displayed if there are no warnings
or additional messages to show

cpapazoglou commented on client/my-sites/plugins/plugin-details-CTA/index.jsx:
```diff
@@ -68,7 +68,10 @@ const PluginDetailsCTA = ( {
-		! isAtomic && ! isJetpack && ( eligibilityHolds || eligibilityWarnings || isEligible );
+		! isAtomic &&
+		! isJetpack &&
+		isEligible &&
```
`isEligible` means `isEligibleForAutomatedTransfer` which isn't the purpose of the `hasEligibilityMessages` check. For sites having `isEligible = false` (it happens to my free site), I don't get the modal and I am redirected to checkout directly. I believe it was mistakenly added there and it worked so long cause it was part of an or `||`. Removing `isEligible` seems working as expected for me.

Your change works fine with a biz or ecom plan with domain, redirecting to checkout directly without showing an empty modal.

epeicher replied:
Good catch! I will remove `isEligible` from the condition and I will test the following scenarios:
- Business plan without custom domain
- Business plan with custom domain
- Free plan

Please let me know if you can think of any other scenarios to test.

cpapazoglou replied:
They seem enough

epeicher committed:
Fix condition that displays eligibility modal

The `isEligible` condition was preventing the display of the eligibility
modal for _free_ sites. That `isEligible` flag is not required as either
the `eligibilityHolds` or the `eligibilityWarnings` arrays will be
populated and will display the modal.

If there are no values in those arrays, the modal does not need to be
displayed anymore.

cpapazoglou reviewed (approved):
Thanks for resolving the feedback. LGTM!
